default/python/AI/InvasionAI.py

```diff
@@ -67,7 +70,35 @@ def get_invasion_fleets():
             avail_pp = el.data()
             for pid in el.key():
                 available_pp[pid] = avail_pp
+        # For planning base trooper invasion targets we have a two-pass system.  (1) In the first pass we consider all
+        # the invasion targets and figure out which ones appear to be suitable for using base troopers against (i.e., we
+        # already have a populated planet in the same system that could build base troopers) and we have at least a
+        # minimal amount of PP available, and (2) in the second pass we go through the reserved base trooper target list
+        # and check to make sure that there does not appear to be too much military action still needed before the
+        # target is ready to be invaded, we double check that not too many base troopers would be needed, and if things
+        # look clear then we queue up the base troopers on the Production Queue and keep track of where we are building
+        # them, and how many; we may also disqualify and remove previously qualified targets (in case, for example,
+        # we lost our base trooper source planet since it was first added to list).
+        #
+        # For planning and tracking base troopers under construction, we use a dictionary store in
+        # foAI.foAIstate.qualifyingTroopBaseTargets, keyed by the invasion target planet ID.  We only store values
+        # for invasion targets that appear likely to be suitable for base trooper use, and store a 2-item list.
+        # TODO: either actually use the first value to help sort out potential mission collisions in
+        # assign_invasion_fleets_to_invade(), or else simplify to a 1-item (source PID) entry
+        # TODO: or consider using a small class with named fields instead of this 2-item list, for clarity
```

Comment:
Speaking longterm, we should hide any such implementation details and use human-readable interface functions instead, e.g. ```register_basetrooper_target(target, troop_count)```. Beyond the scope of this PR of course.